### PR TITLE
misc: Fix SDL_OpenURL on newer iOS releases.

### LINKED ISSUE #11728: iOS SDL_OpenURL() doesn't work
Upon calling SDL_OpenURL() on iOS 18.2, I get:

> BUG IN CLIENT OF UIKIT: The caller of UIApplication.openURL(_:) needs to migrate to the non-deprecated 
> UIApplication.open(_:options:completionHandler:). Force returning false (NO).

### PR DESCRIPTION
**Pull request by icculus**
Apparently as of iOS 18.2, the deprecated API we were using just refuses to work at all.

Fixes #11728.

### CODE REVIEW

File: src/misc/ios/SDL_sysurl.m [hunk 1]
```diff
@@ -35,7 +35,15 @@ bool SDL_SYS_OpenURL(const char *url)
 #else
         NSString *nsstr = [NSString stringWithUTF8String:url];
         NSURL *nsurl = [NSURL URLWithString:nsstr];
-        return [[UIApplication sharedApplication] openURL:nsurl];
+        if (![[UIApplication sharedApplication] canOpenURL:nsurl]) {
+            return SDL_SetError("No handler registerd for this type of URL");
+        }
+        if (@available(iOS 10.0, *)) {
```
Comment (slouken) [inline src/misc/ios/SDL_sysurl.m:41]: You need a tvOS availability here.

Comment (icculus) [reply src/misc/ios/SDL_sysurl.m:41]: Whoops, I thought the `#ifdef` handled this, but it's for VisionOS...how the heck does VisionOS not let you pop up a web browser window? Even Steam can do that in a fullscreen game!  lol

Comment (icculus) [reply src/misc/ios/SDL_sysurl.m:41]: Oh, VisionOS can, it just removed the deprecated API from the start. It has the newer one.

Comment (icculus) [reply src/misc/ios/SDL_sysurl.m]: Pushed a fix in 3bd759e420df3ec7acb2bb5ea37ca1ede9c2f6be, but I have no idea if this is right.

**Pull request merged**